### PR TITLE
Rename addons to tools

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Follow-on action from #3054 

## Development notes

**Note: Ignore failing unit-tests.** Unit tests will fail until [changes go through on kedro-starters](https://github.com/kedro-org/kedro-starters/pull/189), it's a fun circular dependency 🔁 

~~Blocked by :~~
- [x] #3358  

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/framework/cli/starters.py [hunk 1]
```diff
@@ -170,13 +170,13 @@ def starter():
 @click.option("--starter", "-s", "starter_alias", help=STARTER_ARG_HELP)
 @click.option("--checkout", help=CHECKOUT_ARG_HELP)
 @click.option("--directory", help=DIRECTORY_ARG_HELP)
-@click.option("--addons", "-a", "selected_add_ons_flag", help=ADDON_ARG_HELP)
+@click.option("--tools", "-a", "selected_tools", help=TOOLS_ARG_HELP)
```
Comment (merelcht) [inline kedro/framework/cli/starters.py:173]:
```suggestion
@click.option("--tools", "-t", "selected_tools", help=TOOLS_ARG_HELP)
```

Comment (AhdraMeraliQB) [reply kedro/framework/cli/starters.py:173]: Good catch! I'll be going through those next

File: kedro/templates/project/prompts.yml [hunk 2]
```diff
@@ -25,10 +25,10 @@ add_ons:
     6) PySpark: Configuration for working with PySpark
     7) Kedro-Viz: Kedro's native visualisation tool
 
-    Which add-ons would you like to include in your project? [1-7/1,3/all/none]:
+    Which tools would you like to include in your project? [1-7/1,3/all/none]:
   regex_validator: "^(all|none|(( )*\\d*(,\\d*)*(,( )*\\d*)*( )*|( )*((\\d+-\\d+)|(\\d+ - \\d+))( )*))$"
   error_message: |
-    Invalid input. Please select valid options for add-ons using comma-separated values, ranges, or 'all/none'.
+    Invalid input. Please select valid options for tools using comma-separated values, ranges, or 'all/none'.
```
Comment (AhdraMeraliQB) [inline kedro/templates/project/prompts.yml:31]:
```suggestion
    Invalid input. Please select valid options for project tools using comma-separated values, ranges, or 'all/none'.
```

**Review by merelcht — APPROVED** [review]
There's still mentions of `add-ons` in the RELEASE.md but otherwise LGTM 👍

**Review by ankatiyar — APPROVED** [review]
💯

File: kedro/framework/cli/starters.py [hunk 3]
```diff
@@ -466,33 +466,33 @@ def _get_extra_context(  # noqa: PLR0913
     return extra_context
 
 
-def _convert_addon_names_to_numbers(selected_add_ons_flag: str | None) -> str | None:
-    """Prepares add-on selection from the CLI input to the correct format
+def _convert_tools_names_to_numbers(selected_tools: str | None) -> str | None:
```
[the remark this answers is not in the document]
Comment (AhdraMeraliQB) [reply kedro/framework/cli/starters.py]: Changed in [d138efc](https://github.com/kedro-org/kedro/pull/3357/commits/d138efcf4772c04ad5ea2d7bf785d91d05ced835)

**Pull request merged**